python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -229,24 +244,36 @@ def isType(_obj, type_or_seq):
         return isinstance(obj, bare_type_or_seq)
 
 def getName(_obj):
-    """Strip any proxy and then return an objects name"""
+    """Strip any proxy and then return an objects name
+    Args:
+        _obj (object): This may be an instance or a class"""
     obj = stripProxy(_obj)
     returnable = _getName(obj)
     return returnable
 
 def stripProxy(obj):
-    """Removes the proxy if there is one"""
+    """Removes the proxy if there is one
+    Args:
+        obj (object): This may be an instance or a class
+    """
     return getattr(obj, implRef, obj)
 
 
 def addProxy(obj):
-    """Adds a proxy to a GangaObject"""
+    """Adds a proxy to a GangaObject instance or class
+    Args:
+        obj (GangaObject, class): This may be a Ganga object or class which you're wanting to add a proxy to
+    """
     if isType(obj, GangaObject):
         if not isProxy(obj):
             if hasattr(obj, proxyObject):
                 return getattr(obj, proxyObject)
             else:
                 return GPIProxyObjectFactory(obj)
+    elif isclass(obj):
+        if not issubclass(obj, GPIProxyObject):
+            addProxyClass(obj)
```

Comment:
Does this need the `addProxyClass()` call. Is that not done automatically by `getProxyClass()` anyway.